connectors/sources/sharepoint.py

```diff
@@ -339,8 +230,7 @@ async def _api_call(self, url_name, url="", **url_kwargs):
                     exception,
                     ServerDisconnectedError,
                 ):
-                    await self.session.close()
-                    await self._generate_session()
+                    await self.close_session()
```

Comment:
Why close the session with every API call? Seems like you'd want to keep the session open if you're going to be making more api calls.

Reply:
This will only call when `ServerDisconnectedError` error occurs.

Reply:
🤦 totally misread this. Thanks for the correction.